### PR TITLE
[TLI][NFC] Fix ordering of ArmPL and SLEEF tests

### PR DESCRIPTION
**Pull request by mgabka**
This patch sorts the tests which check if SLEEF and ArmPL mappings are used, in the order of the math functions base names.

### CODE REVIEW

File: llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll [hunk 1]
```diff
@@ -2128,25 +2276,98 @@ define void @copysign_f32(ptr noalias %in.ptr, ptr noalias %out.ptr) {
...
 }
 
-declare double @fdim(double, double)
-declare float @fdimf(float, float)
+declare void @sincos(double, ptr, ptr)
+declare void @sincosf(float, ptr, ptr)
 
-define void @fdim_f64(ptr noalias %in.ptr, ptr noalias %out.ptr) {
-; SLEEF-NEON-LABEL: define void @fdim_f64
+define void @test_sincos(ptr noalias %a, ptr noalias %b, ptr noalias %c) {
```
Comment (paschalis-mpeis) [inline llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll:2282]: Similarly with `modf`, could also rename `test_sincos`,  `test_sincosf` to `sincos_f64`,  `sincos_f32`.

**Review by paschalis-mpeis — COMMENTED** [review]
In `replace-with-veclib-*.ll` files, `frem` is at the end of the tests.
It could stay there, if we see it as an exception, since it's the only non-call instruction.

Just wanted to make sure that this did not stay there by mistake.

File: llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll (hunk 1 above)
Comment (mgabka) [reply llvm/test/Transforms/LoopVectorize/AArch64/veclib-function-calls.ll:2282]: I can change it as part of this patch.

**Comment by mgabka** [comment]
> In `replace-with-veclib-*.ll` files, `frem` is at the end of the tests. It could stay there, if we see it as an exception, since it's the only non-call instruction.
> 
> Just wanted to make sure that this did not stay there by mistake.

yes, that was my idea, since frem is the only one instruction, and there is no mappings for it (but jut for fmod) I wanted to leave it at the end.

**Review by paschalis-mpeis — APPROVED** [review]
Thanks for the changes. LGTM.

**Pull request merged**